typings/markup.d.ts

```diff
@@ -0,0 +1,172 @@
+import * as tt from './telegram-types.d'
+
+interface Button {
```

Comment:
Each of these buttons should be exported in order to use the interface externally.

Super handy when creating keyboards like this:
```ts
const buttons: UrlButton[][] = []
buttons.push(…)
Markup.inlineKeyboard(buttons)
```

Reply:
Wouldn't the union of them all be enough?

Reply:
Being more specific is often helpful.
For example: UrlButtons are kept when a message is forwarded. If there are other Buttons than UrlButtons within the same keyboard, all buttons are dropped when forwarded.

I like to use a typed language to have specific types so the compiler can prevent me doing mistakes later on. The use cases for these are often not clear when designing a library so the user of the library should have as many options available as possible. It shouldn't prevent the users doing what they like to do unless its not possible for some reason.

Reply:
Alright. I'm fine with them being exported from `markup.d.ts`, but not `index.d.ts`.

Reply:
How about exporting this from `index.d.ts`?

```ts
export namespace KeyboardButton {
  export interface Text {...}
  export interface ContactRequest {...}
  export interface LocationRequest {...}
  export interface PollRequest {...}
}
export type KeyboardButton =
  | KeyboardButton.Text
  | KeyboardButton.ContactRequest
  | KeyboardButton.LocationRequest
  | KeyboardButton.PollRequest
  | string
```

All union members would be available but wouldn't clutter suggestions.